clang/include/clang/CIR/Dialect/IR/CIRTypes.td

```diff
@@ -139,6 +143,26 @@ class SIntOfWidths<list<int> widths>
         "::mlir::cir::IntType"
     > {}
 
+// A type constraint that allows integer type whose width is among the specified
+// list of possible widths. The signedness of the integer type is not
+// constrained.
+class IntOfWidths<list<int> widths>
+  : Type<And<[
+            CPred<"$_self.isa<::mlir::cir::IntType>()">,
+            Or<!foreach(
+              w, widths,
+              CPred<"$_self.cast<::mlir::cir::IntType>().getWidth() == " # w>
+            )>
+        ]>,
+        !interleave(!foreach(w, widths, w # "-bit"), " or ") #
+            " signed or unsigned int",
+        "::mlir::cir::IntType"
+    > {}
+
+def StdUInt : UIntOfWidths<[8, 16, 32, 64]>;
+def StdSInt : SIntOfWidths<[8, 16, 32, 64]>;
+def StdInt : IntOfWidths<[8, 16, 32, 64]>;
```

Comment:
We already have definitions for a bunch of those, please reuse or merge, but avoid duplicating logic. Example, StdSInt could be:
```
def StdSInt: AnyTypeOf<[SInt8, SInt16, SInt32, SInt64]>;
```
Not tied to a particular approach, but we should think about these things holistically. Also don't add any definiton that isn't used at the moment (e.g. `StdUInt`).

Reply:
There are probably existing ones that aren't used, feel free to clean them up as part of this (if you want)

Reply:
Updated. `UIntOfWidths` and `SIntOfWidths` are not used so I removed them.